### ISSUE TITLE
Error on creating a VM: Caught vmodl fault : Invalid datastore path '[DS2856_SHARED_SILVER_VNX_208] vm-name'.

### ISSUE THREAD

**Issue by 3pr0mpt** [vmware/pyvmomi#350]
Hello all,

We're currently facing a strange issue with pyvmomi over three different datacenter. VM creation is based on the [create_random_marvel_vms.py](https://github.com/vmware/pyvmomi-community-samples/blob/master/samples/create_random_marvel_vms.py) script. 

This is the code we've been using: 
```
task = destfolder.CreateVM_Task(config=configvm, pool=ds.config.entity)
tasks.wait_for_tasks(service_instance, [task])
```
On datacenter 1 and 2, everything is smooth and running fine. On the third datacenter (with exactly the same code), We face the following error:
```
Caught vmodl fault : Invalid datastore path '[DS2856_SHARED_SILVER_VNX_208] vm-name'.
```
The problem could be from the ESX? Maybe, but all shares the exact same API version:

**DC1: WORKS FINE**
Full name: VMware ESXi 5.1.0 build-1065491
Version: 5.1.0
API Version: 5.1

**DC2: WORKS FINE**
Full name: VMware ESXi 5.1.0 build-2323236
Version: 5.1.0
API Version: 5.1

**DC3: _DOESN'T WORK _**
Full name: VMware ESXi 5.1.0 build-1900470
Version: 5.1.0
API Version: 5.1

The Datastore on DC3 has the following Summary:
```
-------> (vim.Datastore.Summary) {
   dynamicType = <unset>,
   dynamicProperty = (vmodl.DynamicProperty) [],
   datastore = 'vim.Datastore:datastore-146235',
   name = 'DS2856_SHARED_SILVER_VNX_208',
   url = 'ds:///vmfs/volumes/537b56e4-1e68d138-af6b-0017a4770066/',
   capacity = 10734733885440L,
   freeSpace = 2385101455360L,
   uncommitted = 193221914624L,
   accessible = true,
   multipleHostAccess = true,
   type = 'VMFS',
   maintenanceMode = 'normal'
}
```
On the vcenter logs, we found the following:
```
2016-02-09T12:26:24.290+01:00 [25928 info 'commonvpxLro' opID=7cd6d19b] [VpxLRO] -- BEGIN task-internal-6407132 --  -- vmodl.query.PropertyCollector.retrievePropertiesEx -- c9462fd4-c987-8158-a2b9-47678ce6ab19(52719c15-a71d-dcc7-dccf-9cf4a7f7b2ae)
2016-02-09T12:26:24.293+01:00 [08580 info 'commonvpxLro' opID=7560b34c] [VpxLRO] -- FINISH task-1040702 -- group-v270961 -- vim.Folder.createVm -- 
2016-02-09T12:26:24.293+01:00 [08580 info 'Default' opID=7560b34c] [VpxLRO] -- ERROR task-1040702 -- group-v270961 -- vim.Folder.createVm: vim.fault.InvalidDatastorePath:
--> Result:
--> (vim.fault.InvalidDatastorePath) {
-->    dynamicType = <unset>, 
-->    faultCause = (vmodl.MethodFault) null, 
-->    datastore = <unset>, 
-->    name = <unset>, 
-->    datastorePath = "[DS2856_SHARED_SILVER_VNX_208] vm-name", 
-->    msg = "", 
--> }
--> Args:
--> 
```

I've looking into multiple sites, as VMWARE KB, Nova source code (and issues on openstack), pyvmomi-community-samples and of course here. But I didn't found anything. There is already someone with this kind of problem using pyvmomi? There is any insider from vmware or advanced user who can help me to troubleshoot what's wrong?

Many thanks in advance.

**Comment by chinmayb** [comment]
its not a problem with pyvmomi ! the example you took, takes 0th datacenter. If you have mulitple datacenters  you need to change the code. 

[https://github.com/vmware/pyvmomi-community-samples/blob/master/samples/create_random_marvel_vms.py#L169](url)

you can modify that example to take datacenter name as a parameter, everything will work :)

**Comment by 3pr0mpt** [comment]
Hello @chinmayb,
Thanks for replying. In fact, everything is already under variables. Maybe I wasn't clear enough when I raised this incident.  But we already had a full automated system using multiple datacenters over multiple datastores. Datacenter and datastore are already provided as input and managed by Python code. 

However, after reading your reply I started to investigate my code (as the datastore object was already provided). At the end it seems that a bug infiltrated my code... It did work in other two DC because all used datastore clusters for VM deployments were the LAST of a list of 5/6 DS... So, they used the last DS object and did worked... But the on DC3, it didn't worked because the DS wasn't the last. I fixed my code and everything works like a charm.

Sorry for wasting your time.

Kind regards.
